### PR TITLE
Improve a11y lint

### PR DESCRIPTION
**Pull request by LMS007**
Small PR to land some of the ease fixes.

- We can't have click handlers on non-interactive elements, but turns out, I don't think we even need this click handler at all because `<Menu>` a distant relative of `<GroupListItem>` has a useElementShouldClose(). Once the Menu opens again, the `isExpanded` prop is reset to `false` so it works just the same. I also tested the screen readers and they don't change as a result of removing the collapseSubmenu method.

- The icon in group-list.js did have not any value in being read out and just made it more confusing so I treat it like a decorative alt tag (blank string)

Fixes some of https://github.com/hypothesis/client/issues/1727

### CODE REVIEW

File: src/sidebar/components/group-list.js [hunk 1]
```diff
@@ -63,11 +63,10 @@ function GroupList({ serviceUrl, settings }) {
     const icon = focusedGroup.organization.logo;
     label = (
       <span className="group-list__menu-label">
-        {/* FIXME-A11Y */}
-        {/* eslint-disable-next-line jsx-a11y/alt-text */}
         <img
           className="group-list__menu-icon"
           src={icon || publisherProvidedIcon(settings)}
+          alt=""
```
Comment (lyzadanger) [inline src/sidebar/components/group-list.js]: Is it possible to provide a meaningful value for the `alt` attr?

File: src/sidebar/components/group-list-item.js [hunk 2]
```diff
@@ -89,9 +86,7 @@ function GroupListItem({
       onToggleSubmenu={toggleSubmenu}
       submenu={
         <Fragment>
-          {/* FIXME-A11Y */}
-          {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions, jsx-a11y/click-events-have-key-events */}
-          <ul onClick={collapseSubmenu}>
+          <ul>
```
Comment (robertknight) [inline src/sidebar/components/group-list-item.js]: @LMS007 - you're right that there isn't any observable behavior change as a result of removing the `onClick` handler, but the reason is not that the `Menu` component handles this, but that the top level `GroupList` component does:

https://github.com/hypothesis/client/blob/7de51cfa2c7ea1bc56d4fb1d24fc9d64555bbec3/src/sidebar/components/group-list.js#L93-L99

If you remove the `onOpenChanged` prop, you'll see that if you expand the menu for a group and then click one of the items in it, that group will still be expanded the next time that the menu is opened. From memory, I think the `onOpenChanged` handler was added after the `onClick={collapseMenu}` logic and I hadn't realized that it made the existing code obsolete.

So to summarize, the change here makes sense, but I wanted to note the correct explanation for posterity.

File: src/sidebar/components/group-list.js [hunk 3]
```diff
@@ -61,13 +62,15 @@ function GroupList({ serviceUrl, settings }) {
   let label;
   if (focusedGroup) {
     const icon = focusedGroup.organization.logo;
+    // If there is no organization name, use a blank string
```
[the remark this answers is not in the document]
Comment (LMS007) [reply src/sidebar/components/group-list.js]: I think you are right, undefined is okay or just gets treated the same as the empty string, but I was trying to be explicit with the html standard for the alt attr.
I'll update the comment

File: src/sidebar/components/group-list.js [hunk 4]
```diff
@@ -5,6 +5,7 @@ import propTypes from 'prop-types';
 import serviceConfig from '../service-config';
 import useStore from '../store/use-store';
 import { isThirdPartyUser } from '../util/account-id';
+import { orgName } from '../util/group-list-item-common';
```
[the remark this answers is not in the document]
Comment (LMS007) [reply src/sidebar/components/group-list.js]: https://github.com/hypothesis/client/issues/1754

File: src/sidebar/components/group-list-item.js (hunk 2 above)
Comment (LMS007) [reply src/sidebar/components/group-list-item.js]: gotcha! thanks Rob

**Comment by LMS007** [comment]
hmm, commit history is not right, looking into it

File: src/sidebar/components/group-list.js [hunk 5]
```diff
@@ -61,13 +62,15 @@ function GroupList({ serviceUrl, settings }) {
   let label;
   if (focusedGroup) {
     const icon = focusedGroup.organization.logo;
+    const altName = orgName(focusedGroup) ? orgName(focusedGroup) : '';
+    // If org name is missing, then treat this icon like decoration
```
Comment (lyzadanger) [inline src/sidebar/components/group-list.js]: Can this comment be above the line it references?
